PWGHF/D2H/TableProducer/dataCreatorDV0Reduced.cxx

```diff
@@ -0,0 +1,393 @@
+            massD = MassDPlus;
+            pVecD = candD.pVector();
+            secondaryVertexD[0] = candD.xSecondaryVertex();
+            secondaryVertexD[1] = candD.ySecondaryVertex();
+            secondaryVertexD[2] = candD.zSecondaryVertex();
+            prongIdsD[0] = candD.prong0Id();
+            prongIdsD[1] = candD.prong1Id();
+            prongIdsD[2] = candD.prong2Id();
+            v0_type = V0_type::K0s;
+            d_type = candD.sign() * D_type::Dplus;
+            massV0 = MassK0Short;
+            break;
+
+          case DecayChannel::Xc3055ToDplusLambda:
+            invMassD = hfHelper.invMassDplusToPiKPi(candD);
+            massD = MassDPlus;
+            pVecD = candD.pVector();
+            secondaryVertexD[0] = candD.xSecondaryVertex();
+            secondaryVertexD[1] = candD.ySecondaryVertex();
+            secondaryVertexD[2] = candD.zSecondaryVertex();
+            prongIdsD[0] = candD.prong0Id();
+            prongIdsD[1] = candD.prong1Id();
+            prongIdsD[2] = candD.prong2Id();
+            v0_type = candD.sign() * V0_type::Lambda;
+            d_type = candD.sign() * D_type::Dplus;
+            massV0 = MassLambda0;
+            break;
+
+          default:
+            LOG(warning) << "Decay channel not valid please choose between Ds1ToDstarK0s, DsStar2ToDplusK0s, Xc3055ToDplusLambda";
+            break;
+        } // switch
+      }   // else if
+
+      // Loop on V0 candidates
+      for (const auto& v0 : V0s) {
+        // propagate V0 to primary vertex
+        std::array<float, 3> pVecV0 = {v0.px(), v0.py(), v0.pz()};
+        if (propagateV0toPV) {
+          std::array<float, 3> pVecV0Orig = {v0.px(), v0.py(), v0.pz()};
+          std::array<float, 3> posVecV0 = {v0.x(), v0.y(), v0.z()};
+          gpu::gpustd::array<float, 2> dcaInfo;
+          auto trackParK0 = o2::track::TrackPar(posVecV0, pVecV0Orig, 0, true);
+          trackParK0.setPID(o2::track::PID::K0);
+          trackParK0.setAbsCharge(0);
+          o2::base::Propagator::Instance()->propagateToDCABxByBz({collision.posX(), collision.posY(), collision.posZ()}, trackParK0, 2.f, matCorr, &dcaInfo);
+          getPxPyPz(trackParK0, pVecV0);
+        }
+
+        float ptV0 = sqrt(pVecV0[0] * pVecV0[0] + pVecV0[1] * pVecV0[1]);
+        auto invMass2DV0 = RecoDecay::m2(std::array{pVecD, pVecV0}, std::array{massD, massV0});
+        // LOG(info) << "V0 p before propagation: " << pVecV0Orig[0] << "," << pVecV0Orig[1] << "," << pVecV0Orig[2];
+        // LOG(info) << "V0 p after propagation: " << pVecV0[0] << "," << pVecV0[1] << "," << pVecV0[2];
+
+        // fill histos
+        registry.fill(HIST("hPtV0"), ptV0);
+        registry.fill(HIST("hV0_type"), v0_type);
+
+        switch (DecayChannel) {
+          case DecayChannel::Ds1ToDstarK0s:
+            registry.fill(HIST("hMassK0s"), v0.mK0Short());
+            registry.fill(HIST("hMassDs1"), sqrt(invMass2DV0) - invMassD);
```

Comment:
The `deltaMass` does not work like this, because the mass resolution of the D does not cancel. You should either use the invariant mass of the D in the DV0 invariant mass, or here you simply rescale by the PDG mass of the D